platforms/android/library/src/main/java/io/element/android/wysiwyg/inputhandlers/InterceptInputConnection.kt

```diff
@@ -125,8 +129,12 @@ internal class InterceptInputConnection(
         val result = processTextEntry(text, start, end)
 
         return if (result != null) {
-            replaceAll(result.text, compositionStart = end, compositionEnd = end)
-            setSelectionOnEditable(editable, result.selection.last, result.selection.last)
+            beginBatchEdit()
+            replaceAll(result.text)
+            val editorSelectionIndex = editorIndex(result.selection.last, editable)
+            setSelection(editorSelectionIndex, editorSelectionIndex)
+            setComposingRegion(editorSelectionIndex, editorSelectionIndex)
+            endBatchEdit()
```

Comment:
There are now a few blocks similar to this. Would it be possible/worthwhile to create a function so that we can write:

```kotlin
val result = processTextEntry(text, start, end)

handleEditorResult(result)
```

Reply:
I'm not sure. The strategies on those different places are different: 

- `setComposingText` applies the selection with the `end` part of the resulting composer update selection, but sets the composition in a custom way.
- `commitText` also applies the selection with the `end` update selection, and the composition will match it, to allow starting a new one at that point.
- `deleteSurroundingText` applies the `start` update selection instead, using it also for the composition.

I don't see a good way to extract a reusable function here, since each action needs different strategies for updating the different text, selection and composition values.